scripts/bump-oss-version.js

```diff
@@ -46,10 +46,34 @@ let versionMajor = branch.slice(0, branch.indexOf(`-stable`));
 // e.g. 0.33.1 or 0.33.0-rc4
 let version = argv._[0];
 if (!version || version.indexOf(versionMajor) !== 0) {
-  echo(`You must pass a tag like ${versionMajor}.[X]-rc[Y] to bump a version`);
+  echo(`You must pass a tag like 0.${versionMajor}.[X]-rc[Y] to bump a version`);
   exit(1);
 }
 
+// Generate version files to detect mismatches between JS and native.
+let [, major, minor, patch, prerelease] = version.match(/^(\d+)\.(\d+)\.(\d+)(?:-rc\.(\d+))?$/);
```

Comment:
Prerelease should be a string (not number) that includes `-beta.1`, etc... can you change the regex to this?
```js
/^(\d+)\.(\d+)\.(\d+)(?:-(.+))?$/
```

Also for clearer error messages:
```js
let match = version.match(...);
if (!match) {
  echo(`You must pass a correctly formatted version; couldn't parse ${version}`);
  exit(1);
}
```